rust/lance-encoding/src/encodings/physical/bitpacking.rs

```diff
@@ -318,28 +314,46 @@ fn bitpack_out_of_line<T: ArrowNativeType + BitPacking>(
         return LanceBuffer::reinterpret_vec(output);
     }
 
-    // If we get here then the last chunk needs to be padded with zeros
-    let remaining_items = data_buffer.len() % ELEMS_PER_CHUNK as usize;
     let last_chunk_start = num_whole_chunks * ELEMS_PER_CHUNK as usize;
+    output.truncate(num_whole_chunks * words_per_chunk);
+    let remaining_items = data_buffer.len() - last_chunk_start;
 
-    let mut last_chunk: Vec<T> = vec![T::from_usize(0).unwrap(); ELEMS_PER_CHUNK as usize];
-    last_chunk[..remaining_items].clone_from_slice(&data_buffer[last_chunk_start..]);
-    let output_start = num_whole_chunks * words_per_chunk;
-    unsafe {
-        BitPacking::unchecked_pack(
-            compressed_bits_per_value,
-            &last_chunk,
-            &mut output[output_start..],
-        );
+    let uncompressed_bits = data.bits_per_value as usize;
+    let tail_bit_savings = uncompressed_bits.saturating_sub(compressed_bits_per_value);
+    let padding_cost = compressed_bits_per_value * (ELEMS_PER_CHUNK as usize - remaining_items);
+    let tail_pack_savings = tail_bit_savings.saturating_mul(remaining_items);
+    let pad_tail = remaining_items > 0 && tail_bit_savings > 0 && padding_cost < tail_pack_savings;
```

Comment:
`remaining_items > 0` and `tail_bit_savings > 0` seems like something we can assert.  I don't think these can ever be zero.